sysl2/sysl/swagger/import.go

```diff
@@ -0,0 +1,319 @@
+		title = si.swagger.Info.Title
+	}
+
+	si.writeLines(fmt.Sprintf("%s %s [package=%s]:", si.args.AppName, quote(title), quote(si.args.Package)))
+	si.ind.Push()
+
+	return nil
+}
+
+func buildQueryString(params []Param) string {
+	query := ""
+	if len(params) > 0 {
+		parts := []string{}
+		for _, p := range params {
+			optional := ""
+			if p.Optional {
+				optional = "?"
+			}
+			parts = append(parts, fmt.Sprintf("%s=%s%s", p.Name, p.Type.Name, optional))
+		}
+		query = " ?" + strings.Join(parts, "&")
+	}
+	return query
+}
+
+func buildRequestBodyString(params []Param) string {
+	body := ""
+	if len(params) > 0 {
+		sort.SliceStable(params, func(i, j int) bool {
+			return strings.Compare(params[i].Name, params[j].Name) < 0
+		})
+		parts := []string{}
+		for _, p := range params {
+			parts = append(parts, fmt.Sprintf("%s <: %s [~body]", p.Name, p.Type.Name))
+		}
+		body = strings.Join(parts, ", ")
+	}
+	return body
+}
+
+func buildRequestHeadersString(params []Param) string {
+	headers := ""
+	if len(params) > 0 {
+		sort.SliceStable(params, func(i, j int) bool {
+			return strings.Compare(params[i].Name, params[j].Name) < 0
+		})
+		parts := []string{}
+		for _, p := range params {
+			optional := map[bool]string{true: "~optional", false: "~required"}[p.Optional]
+
+			safeName := strings.ToLower(strings.ReplaceAll(p.Name, "-", "_"))
+
+			parts = append(parts, fmt.Sprintf("%s <: %s [~header, %s, name=%s]", safeName, p.Type.Name, optional, quote(p.Name)))
+		}
+		headers = strings.Join(parts, ", ")
+	}
+	return headers
+}
+
+func buildPathString(path string, params []Param) string {
+
+	result := path
+
+	for _, p := range params {
+		replacement := fmt.Sprintf("{%s<:%s}", p.Name, p.Type.Name)
+		result = strings.ReplaceAll(result, fmt.Sprintf("{%s}", p.Name), replacement)
+	}
+
+	return result
+}
+
+func (si *swaggerImporter) writeEndpoint(method string, endpoint Endpoint) {
+
+	header := buildRequestHeadersString(endpoint.Params.HeaderParams())
+	body := buildRequestBodyString(endpoint.Params.BodyParams())
+	reqStr := ""
+	if len(header) > 0 && len(body) > 0 {
+		reqStr = fmt.Sprintf(" (%s)", strings.Join([]string{body, header}, ", "))
+	} else if len(header) > 0 || len(body) > 0 {
+		reqStr = fmt.Sprintf(" (%s)", body+header)
+	}
+
+	pathStr := buildPathString(endpoint.Path, endpoint.Params.PathParams())
+
+	si.writeLines(fmt.Sprintf("%s:", pathStr), PushIndent,
+		fmt.Sprintf("%s%s%s:", method, reqStr, buildQueryString(endpoint.Params.QueryParams())), PushIndent,
+		fmt.Sprintf("| %s", getDescription(endpoint.Description)))
+
+	if endpoint.Responses != nil {
+		var outs []string
+		for statusCode, response := range endpoint.Responses.StatusCodeResponses {
+			if schema := response.Schema; schema != nil {
+				outs = append(outs, findReferencedType(*schema))
+			} else {
+				outs = append(outs, fmt.Sprintf("%d", statusCode))
+			}
+
+		}
+		if endpoint.Responses.Extensions != nil {
+			println("x-* responses not implemented")
```

Comment:
It would be great if a descriptive log is shown instead of println

Reply:
done